src/storage-client/src/sink.rs

```diff
@@ -469,55 +466,49 @@ pub async fn determine_latest_progress_record(
                 )
             })?;
 
+        info!("fetching latest progress record for {progress_key}, lo/hi: {lo}/{hi}");
+
         // Empty topic.  Return early to avoid unnecessary call to kafka below.
         if hi == 0 {
             return Ok(None);
         }
 
         let mut latest_ts = None;
         let mut latest_offset = None;
-
-        let progress_key_bytes = progress_key.as_bytes();
         while let Some((key, message, offset)) = get_next_message(progress_client, timeout)? {
-            debug_assert!(offset >= latest_offset.unwrap_or(0));
+            assert!(latest_offset < Some(offset));
             latest_offset = Some(offset);
 
-            let timestamp_opt = if &key == progress_key_bytes {
-                let progress: ProgressRecord = serde_json::from_slice(&message)?;
-                Some(progress.timestamp)
-            } else {
-                None
-            };
-
-            if let Some(ts) = timestamp_opt {
-                if ts >= latest_ts.unwrap_or_else(timely::progress::Timestamp::minimum) {
-                    latest_ts = Some(ts);
-                }
+            if &key != progress_key.as_bytes() {
+                continue;
             }
 
-            // If the next possible offset for the client is past the high watermark, we've seen
-            // everything we expect to see.
-            let position = progress_client
-                .position()?
-                .find_partition(progress_topic, partition)
-                .ok_or_else(|| anyhow!("No progress info for known partition"))?
-                .offset();
-            if let Offset::Offset(i) = position {
-                if i >= hi {
-                    break;
```

Comment:
Removing this is likely to cause a fairly significant liveness issue... see the commit where it was introduced!